src/brsTypes/components/RoArray.ts

```diff
@@ -5,85 +5,33 @@ import { Callable, StdlibArgument } from "../Callable";
 import { Interpreter } from "../../interpreter";
 import { RoAssociativeArray } from "./RoAssociativeArray";
 
-/**
- * Gives the order for sorting different types in a mixed array
- *
- * Mixed Arrays seem to get sorted in this order (tested with Roku OS 9.4):
- *
- * numbers in numeric order
- * strings in alphabetical order,
- * assocArrays in original order
- * everything else in original order
- */
-function getTypeSortIndex(a: BrsType): number {
-    if (isBrsNumber(a)) {
-        return 0;
-    } else if (isBrsString(a)) {
-        return 1;
-    } else if (a instanceof RoAssociativeArray) {
-        return 2;
-    }
-    return 3;
-}
-
-/**
- * Sorts two BrsTypes in the order that Roku would sort them
- * @param originalArray A copy of the original array. Used to get the order of items
- * @param a
- * @param b
- * @param caseInsensitive Should strings be compared case insensitively? defaults to false
- * @param sortInsideTypes Should two numbers or two strings be sorted? defaults to true
- * @return compare value for array.sort()
- */
-function sortCompare(
-    originalArray: BrsType[],
-    a: BrsType,
-    b: BrsType,
-    caseInsensitive: boolean = false,
-    sortInsideTypes: boolean = true
-): number {
-    let compare = 0;
-    if (a !== undefined && b !== undefined) {
-        const aSortOrder = getTypeSortIndex(a);
-        const bSortOrder = getTypeSortIndex(b);
-        if (aSortOrder < bSortOrder) {
-            compare = -1;
-        } else if (bSortOrder < aSortOrder) {
-            compare = 1;
-        } else if (sortInsideTypes && isBrsNumber(a)) {
-            // two numbers are in numeric order
-            compare = (a as Comparable).greaterThan(b).toBoolean() ? 1 : -1;
-        } else if (sortInsideTypes && isBrsString(a)) {
-            // two strings are in alphabetical order
-            let aStr = a.toString();
-            let bStr = b.toString();
-            if (caseInsensitive) {
-                aStr = aStr.toLowerCase();
-                bStr = bStr.toLowerCase();
-            }
-            // roku does not use locale for sorting strings
-            compare = aStr > bStr ? 1 : -1;
-        } else {
-            // everything else is in the same order as the original
-            const aOriginalIndex = originalArray.indexOf(a);
-            const bOriginalIndex = originalArray.indexOf(b);
-            if (aOriginalIndex > -1 && bOriginalIndex > -1) {
-                compare = aOriginalIndex - bOriginalIndex;
-            }
-        }
-    }
-    return compare;
-}
-
 export class RoArray extends BrsComponent implements BrsValue, BrsIterable {
     readonly kind = ValueKind.Object;
+    private maxSize = 0;
+    private resizable = true;
+    private enumIndex: number;
     elements: BrsType[];
-    enumIndex: number;
 
-    constructor(elements: BrsType[]) {
+    constructor(elements: BrsType[]);
```

Comment:
We have to keep the internal constructor signature that is used by other components and test cases, this is not possible via BrightScript code, only the two params signature is allowed